python/cuml/cuml/cluster/kmeans.pyx

```diff
@@ -209,6 +210,10 @@ class KMeans(UniversalBase,
             params.init = self._params_init
             params.max_iter = <int>self.max_iter
             params.tol = <double>self.tol
+            # After transferring from one device to another `_seed` might not be set
+            # so we need to pass a dummy value here. Its value does not matter as the
+            # seed is only used during fitting
+            params.rng_state.seed = getattr(self, "_seed", 0)
```

Comment:
```suggestion
            params.rng_state.seed = <int>getattr(self, "_seed", 0)
```

I _think_ the CI failure is because we need to cast this to an `int` since `params` is a struct,